substrate/frame/staking-async/src/session_rotation.rs

```diff
@@ -964,7 +964,14 @@ impl<T: Config> EraElectionPlanner<T> {
 		// collect or update the pref of all winners.
 		for stash in &elected_stashes {
 			let pref = Validators::<T>::get(stash);
-			Eras::<T>::set_validator_prefs(new_planned_era, stash, pref);
+			// Only set prefs if they exist (i.e., the stash is actually a validator).
+			if let Some(pref_value) = pref {
+				Eras::<T>::set_validator_prefs(new_planned_era, stash, pref_value);
+			} else {
+				// This case should ideally not happen for an elected winner,
+				// but handle defensively.
+				crate::log!(warn, "Elected validator {:?} has no prefs set.", stash);
```

Comment:
```suggestion
				defensive!(warn, "Elected validator {:?} has no prefs set.", stash);
```

cc @Ank4n as you pointed out, this code path + how we handle `ElectableStashes` is generally very naive :( looking forward to investigate it more before polkadot AHM.

Reply:
I think it's expected, the change to `OptionQuery` revealed that these tests had `probably` an incorrect setup - they were simulating elections with unregistered validators, which should indeed fail now. @kianenigma , @Ank4n  WDYT?